src/domain/orders/details/rma-sub-modals/products.tsx

```diff
@@ -0,0 +1,316 @@
+        ),
+      },
+    ]
+  }, [])
+
+  const {
+    getTableProps,
+    getTableBodyProps,
+    headerGroups,
+    rows,
+    prepareRow,
+    canPreviousPage,
+    canNextPage,
+    pageCount,
+    nextPage,
+    previousPage,
+    // Get the state from the instance
+    state: { pageIndex, pageSize, selectedRowIds },
+  } = useTable(
+    {
+      columns,
+      data: variants || [],
+      manualPagination: true,
+      initialState: {
+        pageIndex: currentPage,
+        pageSize: limit,
+        selectedRowIds: selectedItems.reduce((prev, { id }) => {
+          prev[id] = true
+          return prev
+        }, {}),
+      },
+      pageCount: numPages,
+      autoResetSelectedRows: false,
+      autoResetPage: false,
+      getRowId: (row) => row.id,
+    },
+    usePagination,
+    useRowSelect,
+    (hooks) => {
+      hooks.visibleColumns.push((columns) => [
+        // Let's make a column for selection
+        {
+          id: "selection",
+          // The header can use the table's getToggleAllRowsSelectedProps method
+          // to render a checkbox
+          Header: ({ getToggleAllRowsSelectedProps }) => {
+            return (
+              <div>
+                <IndeterminateCheckbox {...getToggleAllRowsSelectedProps()} />
+              </div>
+            )
+          },
+          // The cell can use the individual row's getToggleRowSelectedProps method
+          // to the render a checkbox
+          Cell: ({ row }) => {
+            return (
+              <div>
+                <IndeterminateCheckbox {...row.getToggleRowSelectedProps()} />
+              </div>
+            )
+          },
+        },
+        ...columns,
+      ])
+    }
+  )
+
+  useEffect(() => {
+    setSelectedVariants((selectedVariants) => [
+      ...selectedVariants.filter(
+        (sv) => Object.keys(selectedRowIds).findIndex((id) => id === sv.id) > -1
+      ),
+      ...(variants?.filter(
+        (v) =>
+          selectedVariants.findIndex((sv) => sv.id === v.id) < 0 &&
+          Object.keys(selectedRowIds).findIndex((id) => id === v.id) > -1
+      ) || []),
+    ])
+  }, [selectedRowIds])
+
+  const handleNext = () => {
+    if (canNextPage) {
+      setOffset((old) => old + pageSize)
+      setCurrentPage((old) => old + 1)
+      nextPage()
+    }
+  }
+
+  const handlePrev = () => {
+    if (canPreviousPage) {
+      setOffset((old) => old - pageSize)
+      setCurrentPage((old) => old - 1)
+      previousPage()
+    }
+  }
+
+  const handleSearch = (q) => {
+    setOffset(0)
+    setCurrentPage(0)
+    setQuery(q)
+  }
+
+  const handleSubmit = () => {
+    onSubmit(selectedVariants)
+    pop()
+  }
+
+  return (
+    <>
+      <Modal.Content isLargeModal={isLargeModal}>
+        <div className="h-full">
+          <h2 className="inter-base-semibold mb-4">Search for additional </h2>
+        </div>
+        <Table enableSearch handleSearch={handleSearch} {...getTableProps()}>
```

Comment:
search placeholder should be "Search Products" also missing filter/sort

also searching makes the modal flicker a lot. It collapses while searching so maybe we should add some min height equal to 12 product rows (as in the [design](https://www.figma.com/file/dlraOnQvQnrHQL0A7Ysbrs/Web-App-2.0?node-id=1507%3A7566)) and then display a Spinner while it is loading